jtune.py

```diff
@@ -1753,6 +1756,41 @@ def _at_exit(raw_gc_log=None, jmap_data=None, jstat_data=None,
                   optimized_for_ygcs_rate)
 
 
+def _generate_rotated_gc_log_name(gc_log_file, log_number):
+    return gc_log_file + "." + str(log_number)
+
+def _get_rotated_log_file(gc_log_file):
+    """Function will scan existing log files to determine latest rotated log, if none found will return
+       non rotated file name.
+    """
+    log_number = 0
+    while os.path.isfile(_generate_rotated_gc_log_name(gc_log_file, log_number)):
+        log_number += 1
+
+    if log_number == 0:
+        logger.debug("\n".join(
+            textwrap.wrap(
+                "Was not able to find a rotated GC log for this process. defaulting to gc log from process.",
+                textwrap_offset)))
+        return gc_log_file
+
+    return _generate_rotated_gc_log_name(gc_log_file, (log_number - 1))
+
+def get_gc_log_file(procdetails=None):
```

Comment:
I'm not sure if I'm 100% consistent here (I try to be at least), but the argument for get_gc_log_file should not be optional, so I'd get rid of the procdetails=None, and make it required instead (esp since you don't check to see if the key exists in line #1780).